kong/db/schema/entities/wasm_filter_chains.lua

```diff
@@ -33,22 +33,36 @@ local filter = {
 return {
   name = "wasm_filter_chains",
```

Comment:
The admin API endpoint for this entity is actually [/wasm/filter-chains](https://github.com/Kong/kong/blob/3afa8c8ae94f93954a333c1e05563d1fbc2a5c33/kong/db/schema/entities/wasm_filter_chains.lua#L37). It was an assumption of mine early on that "maybe we'll have other wasm things, so this seems logical to make wasm a namespace and filter-chains a namespace member," but I'm not married to this endpoint name/structure if we want to change it.

In the context of the Lua DAO/entity definitions, this `name` field decides the string key on the `kong.db` global object (as well as the name of the table in the DB), so using snake case is the convention in order to make code nice (`kong.db.wasm_filter_chains:select()` vs `kong.db["wasm-filter-chains"]:select()`).

Reply:
re: do we need the `wasm` prefix

I don't actually have a strong opinion on this one. When I named this I just went with the most descriptive name I could think of. Maybe we can chat about it at one of the next meetings and see if anyone else has a strong case to make with/without `wasm_`.

Reply:
> actually [/wasm/filter-chains](https://github.com/Kong/kong/blob/3afa8c8ae94f93954a333c1e05563d1fbc2a5c33/kong/db/schema/entities/wasm_filter_chains.lua#L37).

I should have noticed that, we don't have such namespacing for a single entity anywhere before, do we?


> so using snake case is the convention in order to make code nice (kong.db.wasm_filter_chains:select() vs

No objection on using snake casing in Lua code.

> re: do we need the wasm prefix

We have '/plugins' and not '/lua-plugins', right?
I'm not strongly opinionated on this either but lean heavily towards using a generic term since 'wasm' is not something all users will (or must) know when using filters in Kong. Same as all Kong users don't necessary need to know OpenResty.